MyString.java

```diff
@@ -78,8 +112,36 @@ public static String randomStringOfLetters(int n) {
      * @return a string consisting of str1 minus all the characters of str2
      */
     public static String remove(String str1, String str2) {
-       //// Replace the following statement with your code
-        return null;
+       if (str2.isEmpty()) {
+          return str1;
+       }
+
+       if (str1.isEmpty()) {
+          return "";
+       }
+
+       for (int i = 0; i < str2.length(); i++) {
+            int index = str1.indexOf(str2.charAt(i));
+            if (index != -1) {
+                str1 = removeIndex(str1, index);
+                
+            }
+       }
+
+        return str1;
+    }
+
+    public static String removeIndex(String str1, int index) {
+        String newString = "";
+        for (int i = 0; i < index; i++) {
+            newString += str1.charAt(i);
+        }
+
+        for (int j = index +1; j < str1.length(); j++) {
+            newString += str1.charAt(j);
+        }
+
+        return newString;

```

Comment:
nice function, alternative:
```
public static String removeIndex(String str1, int index) {
    return str.substring(0, index) + str.substring(index + 1); 
}
```
you can guard on case where index is last index.. see recitation 7 isAnagram